src/pages/UnlockedWallet/TransfersPage/index.tsx

```diff
@@ -16,19 +16,206 @@ You should have received a copy of the GNU Lesser General Public License
+      : map(selectedDirections, 'label').join(', ')
+
+  const renderAssetsSelectedValue = () =>
+    selectedAssets.length === 0
+      ? ''
+      : selectedAssets.length === assets.length
+      ? 'All selected'
+      : selectedAssets.map((asset) => asset.symbol ?? asset.id).join(', ')
+
+  const resetFilters = () => {
+    setSelectedAddresses(addresses)
+    setSelectedDirections(directionOptions)
+    setSelectedAssets(assets)
+  }
+
+  return (
+    <UnlockedWalletPage
+      title={t('Transfers')}
+      subtitle={t('Browse and download your transaction history. Execute new transfers easily.')}
+      isInfoMessageVisible={!infoMessageClosed}
+      closeInfoMessage={closeInfoMessage}
+      infoMessageLink={links.faq}
+      infoMessage={t('You have questions about transfers ? Click here!')}
+    >
+      <Filters>
+        <FilterTiles>
+          <Tile>
+            <MultiSelect
+              label={t('Addresses')}
+              modalTitle={t('Select addresses')}
+              options={addresses}
+              selectedOptions={selectedAddresses}
+              selectedOptionsSetter={setSelectedAddresses}
+              renderSelectedValue={renderAddressesSelectedValue}
+              getOptionKey={(address) => address.hash}
+              getOptionText={(address) => address.label || address.hash}
+              renderOption={(address: Address) => <SelectOptionAddress address={address} />}
+            />
+          </Tile>
+          <Tile>
+            <MultiSelect
+              label={t('Assets')}
+              modalTitle={t('Select assets')}
+              options={assets}
+              selectedOptions={selectedAssets}
+              selectedOptionsSetter={setSelectedAssets}
+              renderSelectedValue={renderAssetsSelectedValue}
+              getOptionKey={(asset) => asset.id}
+              getOptionText={(asset) => asset.name ?? asset.symbol ?? asset.id}
+              renderOption={(asset) => <SelectOptionAsset asset={asset} />}
+            />
+          </Tile>
+          <Tile>
+            <MultiSelect
+              label={t('Directions')}
+              modalTitle={t('Select directions')}
+              options={directionOptions}
+              selectedOptions={selectedDirections}
+              selectedOptionsSetter={setSelectedDirections}
+              renderSelectedValue={renderDirectionsSelectedValue}
+              getOptionKey={(direction) => direction.value.toString()}
+              getOptionText={(direction) => direction.label}
+            />
+          </Tile>
+        </FilterTiles>
+        <Buttons>
+          <Button role="secondary" short onClick={resetFilters}>
+            {t('Reset filters')}
+          </Button>
+        </Buttons>
+      </Filters>
+      <UnlockedWalletPanel top>
+        <TransactionList
+          addressHashes={map(selectedAddresses, 'hash')}
+          directions={map(selectedDirections, 'value')}
+          assetIds={map(selectedAssets, 'id')}
```

Comment:
Instead of always redefining the simple function that creates an array of a specific property of an objects array, I discovered that `lodash` provides the `map` function!